tests/test_filtsmooth/test_statespace/test_discrete_transition_utils.py

```diff
@@ -45,8 +47,41 @@ def test_cholesky_update(spdmat1, spdmat2):
 
 
 def test_cholesky_optional(spdmat1, test_ndim):
+    """Assert that cholesky_update() transforms a non-square matrix square-root into a
+    correct Cholesky factor."""
     H = np.random.rand(test_ndim, test_ndim)
     expected = np.linalg.cholesky(H @ spdmat1 @ H.T)
     S1 = np.linalg.cholesky(spdmat1)
     received = pnfss.cholesky_update(H @ S1)
     np.testing.assert_allclose(expected, received)
+
+
+def test_triu_to_tril():
+    """triu_to_positive_tril can handle small values on the diagonal.
+
+    This is motivatived by the failing sign change in triu_to_positive_tril
+    due to np.sign assigning 0 to really small values.
+
+    This fix unfortunately cannot be tested reliably, since in artificial
+    setups, np.sign is super well-behaved.
+    Nevertheless, it doesn't hurt to test this lower level function.
```

Comment:
:(

Reply:
But does it even assign 0 to really small values? I would have expected that the error is just that some value on the diagonal was exactly zero, after which we then deleted a whole row (or col, idk).

Reply:
aaah that could've been the issue as well. I dont know. I will remove this text then.